The execution of "make test-docker-py" will fail，If you depend on the proxy to access the Internet.

When I run `make test-docker-py` in my development environment, found that most use cases do not pass,the results are as follows :
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.9 -- py-1.4.31 -- pytest-2.7.2
rootdir: /docker-py, inifile: pytest.ini
plugins: cov
collected 111 items

../../../../../docker-py/tests/integration/api_test.py EEEEEEEEEE
../../../../../docker-py/tests/integration/build_test.py EEEEE
../../../../../docker-py/tests/integration/container_test.py EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEs
../../../../../docker-py/tests/integration/exec_test.py EEEEEEE
../../../../../docker-py/tests/integration/image_test.py EEEEEEEEEs
../../../../../docker-py/tests/integration/network_test.py EEssssEEssEEE
../../../../../docker-py/tests/integration/regression_test.py EEEEEE
../../../../../docker-py/tests/integration/volume_test.py EEEEEEE

==================================== ERRORS ====================================
_________________ ERROR at setup of InformationTest.test_info __________________
/docker-py/tests/integration/conftest.py:30: in setup_test_session
    c.inspect_image(BUSYBOX)
/docker-py/docker/utils/decorators.py:21: in wrapped
    return f(self, resource_id, *args, **kwargs)
/docker-py/docker/api/image.py:143: in inspect_image
    self._get(self._url("/images/{0}/json", image)), True
/docker-py/docker/client.py:150: in _result
    self._raise_for_status(response)
/docker-py/docker/client.py:145: in _raise_for_status
    raise errors.NotFound(e, response, explanation=explanation)
E   NotFound: 404 Client Error: Not Found ("No such image: busybox:buildroot-2014.02")
---------------------------- Captured stderr setup -----------------------------

pulling busybox:buildroot-2014.02
Pulling repository docker.io/library/busybox - None
None - None
   ... ...
```

After examination, I found that when the container is run need to set up a network proxy. Therefore, I modify `DOCKER_FLAGS` in the `Makefile` as follows：
```
DOCKER_FLAGS := docker run --rm -i --privileged $(DOCKER_USERDEF_ENVS) $(DOCKER_ENVS) $(DOCKER_MOUNT)
```
And set the appropriate environment variables in the host
```
export DOCKER_USERDEF_ENVS="-e http_proxy=http://10.10.10.10:80 -e https_proxy=http://10.10.10.10:80 -e no_proxy='127.0.0.1,localhost'"
```

Executed `make test-docker-py` command again, It works! The results are as follows:
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.9 -- py-1.4.31 -- pytest-2.7.2
rootdir: /docker-py, inifile: pytest.ini
plugins: cov
collected 111 items

../../../../../docker-py/tests/integration/api_test.py ..........
../../../../../docker-py/tests/integration/build_test.py .....
../../../../../docker-py/tests/integration/container_test.py ....................................................s
../../../../../docker-py/tests/integration/exec_test.py sssssss
../../../../../docker-py/tests/integration/image_test.py .........s
../../../../../docker-py/tests/integration/network_test.py ..ssss..ss...
../../../../../docker-py/tests/integration/regression_test.py ......
../../../../../docker-py/tests/integration/volume_test.py .......
=========================== short test summary info ============================
SKIP [1] /docker-py/tests/integration/exec_test.py:75: Exec driver not native
SKIP [1] tests/integration/network_test.py:165: API version is too low (< 1.22)
SKIP [1] /docker-py/tests/integration/exec_test.py:43: Exec driver not native
SKIP [1] /docker-py/tests/integration/exec_test.py:27: Exec driver not native
SKIP [1] tests/integration/network_test.py:262: API version is too low (< 1.22)
SKIP [1] tests/integration/network_test.py:238: API version is too low (< 1.22)
SKIP [1] /docker-py/tests/integration/exec_test.py:117: Exec driver not native
SKIP [1] tests/integration/container_test.py:1048: API version is too low (< 1.22)
SKIP [1] /docker-py/tests/integration/exec_test.py:59: Exec driver not native
SKIP [1] tests/integration/network_test.py:214: API version is too low (< 1.22)
SKIP [1] tests/integration/image_test.py:216: Doesn't work inside a container - FIXME
SKIP [1] /docker-py/tests/integration/exec_test.py:11: Exec driver not native
SKIP [1] tests/integration/network_test.py:189: API version is too low (< 1.22)
SKIP [1] tests/integration/network_test.py:129: API version is too low (< 1.22)
SKIP [1] /docker-py/tests/integration/exec_test.py:93: Exec driver not native

=================== 96 passed, 15 skipped in 157.64 seconds ====================
---> Making bundle: .integration-daemon-stop (in bundles/1.12.0-dev/test-docker-py)
++++ cat bundles/1.12.0-dev/test-docker-py/docker.pid
+++ kill 5880
+++ /etc/init.d/apparmor stop
Clearing AppArmor profiles cache:.
All profile caches have been cleared, but no profiles have been unloaded.
Unloading profiles will leave already running processes permanently
unconfined, which can lead to unexpected situations.

To set a process to complain mode, use the command line tool
'aa-complain'. To really tear down all profiles, run the init script
with the 'teardown' option."
```

--- a/Makefile
+++ b/Makefile
@@ -44,7 +44,7 @@ GIT_BRANCH_CLEAN := $(shell echo $(GIT_BRANCH) | sed -e "s/[^[:alnum:]]/-/g")
 DOCKER_IMAGE := docker-dev$(if $(GIT_BRANCH_CLEAN),:$(GIT_BRANCH_CLEAN))
 DOCKER_DOCS_IMAGE := docker-docs$(if $(GIT_BRANCH_CLEAN),:$(GIT_BRANCH_CLEAN))
 
-DOCKER_FLAGS := docker run --rm -i --privileged $(DOCKER_ENVS) $(DOCKER_MOUNT)
+DOCKER_FLAGS := docker run --rm -i --privileged $(DOCKER_USERDEF_ENVS) $(DOCKER_ENVS) $(DOCKER_MOUNT)
 
 # if this session isn't interactive, then we don't want to allocate a
 # TTY, which would fail, but if it is interactive, we do want to attach

--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,13 @@ DOCKER_ENVS := \
 	-e DOCKER_USERLANDPROXY \
 	-e TESTDIRS \
 	-e TESTFLAGS \
-	-e TIMEOUT
+	-e TIMEOUT \
+	-e HTTP_PROXY \
+	-e HTTPS_PROXY \
+	-e NO_PROXY \
+	-e http_proxy \
+	-e https_proxy \
+	-e no_proxy
 # note: we _cannot_ add "-e DOCKER_BUILDTAGS" here because even if it's unset in the shell, that would shadow the "ENV DOCKER_BUILDTAGS" set in our Dockerfile, which is very important for our official builds
 
 # to allow `make BIND_DIR=. shell` or `make BIND_DIR= test`
@@ -44,7 +50,7 @@ GIT_BRANCH_CLEAN := $(shell echo $(GIT_BRANCH) | sed -e "s/[^[:alnum:]]/-/g")
 DOCKER_IMAGE := docker-dev$(if $(GIT_BRANCH_CLEAN),:$(GIT_BRANCH_CLEAN))
 DOCKER_DOCS_IMAGE := docker-docs$(if $(GIT_BRANCH_CLEAN),:$(GIT_BRANCH_CLEAN))
 
-DOCKER_FLAGS := docker run --rm -i --privileged $(DOCKER_USERDEF_ENVS) $(DOCKER_ENVS) $(DOCKER_MOUNT)
+DOCKER_FLAGS := docker run --rm -i --privileged $(DOCKER_ENVS) $(DOCKER_MOUNT)
 
 # if this session isn't interactive, then we don't want to allocate a
 # TTY, which would fail, but if it is interactive, we do want to attach

